packages/react-router/tests/link.test.tsx

```diff
@@ -619,6 +668,64 @@ describe('Link', () => {
     expect(await screen.findByText('UserId: userId')).toBeInTheDocument()
   })
 
+  test('when navigating to /posts with a beforeLoad that throws an error', async () => {
```

Comment:
we would also need a test that shows that the error is bubbling up from e.g. in `posts` to `__root`